src/common/encoding.cpp

```diff
@@ -261,6 +261,76 @@ std::string from_utf8(const std::string& input, eFileEncoding toEncoding)
     return ret;
 }
 
+#else
+
+#include <cerrno>
+#include <vector>
+
+#include <iconv.h>
+
+static const char* get_iconv_encoding_name(eFileEncoding encoding)
+{
+    switch (encoding) {
+    case eFileEncoding::LATIN1:
+        return "ISO-8859-1";
+    case eFileEncoding::SHIFT_JIS:
+        return "CP932";
+    case eFileEncoding::EUC_KR:
+        return "CP949";
+    case eFileEncoding::UTF8:
+        return "UTF-8";
+    }
+    panic("Error", "Incorrect eFileEncoding");
+    abort();
+    return "";
+}
+
+static std::string convert(const std::string& input, eFileEncoding from, eFileEncoding to)
+{
+    const auto* source_encoding_name = get_iconv_encoding_name(from);
+    const auto* target_encoding_name = get_iconv_encoding_name(to);
+
+    auto* icd = iconv_open(target_encoding_name, source_encoding_name);
```

Comment:
Not only would this be a leak because of an early return, I accidentally removed the `iconv_close()` call at some point, so this *always* leaks.

Reply:
29adb5919ec96695f32b3d878fb0304f0cfd7b74